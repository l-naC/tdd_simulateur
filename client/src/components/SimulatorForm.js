import React, { useState, Component } from "react";
import {
  Container,
  CssBaseline,
  TextField,
  Link,
  Grid,
  Button,
  Typography,
  Box,
} from "@material-ui/core";
import axios from "axios";

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright © "}
      <Link color="inherit" href="/">
        Léna Clavier &amp; Valentine Lav
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

const styles = (theme) => ({
  selectEmpty: {
    /* marginTop: spacing(2), */
  },
  paper: {
    /* marginTop: spacing(8), */
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    /* margin: spacing(1), */
    /* backgroundColor: palette.secondary.main, */
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    /* marginTop: spacing(3), */
  },
  submit: {
    /* margin: spacing(3, 0, 2), */
  },
});

export default class SimulatorForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      birthYear: "",
      ageWork: "",
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
    console.log(event);
  }

  handleSubmit(event) {
    event.preventDefault();
    console.log(this.state);
    axios
      .get("http://localhost:9000/simulator", {
        params: {
          birthYear: this.state.birthYear,
          ageWork: this.state.ageWork,
        },
      })
      .then((res) => {
        this.setState({ age: res.data.count });
        console.log("test", this.state.age);
        console.log("succes", res.data);
      })
      .catch((e) => {
        console.error("error", e);
      });
  };

  render() {
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className="">
          <br />
          <br />
          <Typography component="h1" variant="h5">
            Simulateur de retraite
          </Typography>
          <Typography component="p">
            Savoir à quel âge on peut partir avec une retraite à taux plein
            (avec une carrière continue)
          </Typography>
          <br />
          <br />
          <form onSubmit={this.handleSubmit} noValidate>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  required
                  name="birthYear"
                  label="Année de naissance"
                  id="birthYear"
                  variant="outlined"
                  placeholder="AAAA"
                  value={this.state.birthYear}
                  onChange={this.handleChange}
                ></TextField>
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  name="ageWork"
                  label="Âge d'entrée dans la vie active"
                  id="ageWork"
                  variant="outlined"
                  placeholder=""
                  value={this.state.ageWork}
                  onChange={this.handleChange}
                />
              </Grid>
            </Grid>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <Button
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                >
                  Calculer mon âge de retraite
                </Button>
              </Grid>
            </Grid>
          </form>
        </div>
        <br />
        <br />
        <div>
          <Typography component="p">
            {this.state.age
              ? "Votre âge de retraite à taux plein est " + this.state.age
              : ""}
          </Typography>
        </div>
        <Box mt={5}>
          <Copyright />
        </Box>
      </Container>
    );
  }
}
