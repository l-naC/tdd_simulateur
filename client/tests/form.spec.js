import React from "react";
import renderer from "react-test-renderer";
import SimulatorForm from "./../src/components/SimulatorForm"

test("Form rendering", () => {
  const component = renderer
    .create(
      <SimulatorForm></SimulatorForm>
    );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//let wrapper;

// test("Birth year check", () => {
//     wrapper = shallow(<SimulatorForm />);
//   wrapper
//     .find('input[type="text"]')
//     .simulate("change", {
//       target: { name: "username", value: "krishankantsinghal" },
//     });
//   expect(wrapper.state("username")).toEqual("krishankantsinghal");
// });
