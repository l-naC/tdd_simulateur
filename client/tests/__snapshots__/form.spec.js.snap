// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form rendering 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className=""
  >
    <br />
    <br />
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Simulateur de retraite
    </h1>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Savoir à quel âge on peut partir avec une retraite à taux plein (avec une carrière continue)
    </p>
    <br />
    <br />
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="birthYear"
              id="birthYear-label"
            >
              Année de naissance
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="birthYear"
                name="birthYear"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="AAAA"
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Année de naissance
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="ageWork"
              id="ageWork-label"
            >
              Âge d'entrée dans la vie active
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="ageWork"
                name="ageWork"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder=""
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Âge d'entrée dans la vie active
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Calculer mon âge de retraite
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
  <br />
  <br />
  <div>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-5"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Léna Clavier & Valentine Lav
      </a>
       
      2021
      .
    </p>
  </div>
</main>
`;
